### DISCUSSION TITLE
Anyway to override subdependency of a dependency?

### DISCUSSION THREAD

**Discussion by tigerinus** [pyinstaller/pyinstaller#8281] [Help] [answered]
The following issue is found in our code packed by Pyinstaller:

- https://github.com/Unstructured-IO/unstructured/issues/2503

Basically I'd like to override dependency C of dependency B of my code A.

Although it's not a Pyinstaller issue, I wonder if there is anyway to workaround it?

Thanks.

**Comment by rokm** [comment, marked as answer]
As far as PyInstaller is concerned, if you want `opencv-python-headless` collected instead of `opencv-python`, you should ensure that the former is available/installed in the build environment instead of the latter.

I.e., you could remove `opencv-python` and install `opencv-python-headless` after installing whatever pulled `opencv-python` in, and before running PyInstaller:

```
...
pip uninstall opencv-python
pip install opencv-python-headless
...
```
